### PR TITLE
Add requested version to hostfxr_resolve_sdk2

### PR DESCRIPTION
**Pull request by elinor-fung**
Add `hostfxr_resolve_sdk2_result_key_t::requested_version` for indicating the SDK version requested (if any). The callback will be invoked with that key only if there was a version specified.

### CODE REVIEW

File: src/native/corehost/fxr/sdk_resolver.cpp [hunk 1]
```diff
@@ -40,11 +40,16 @@ sdk_resolver::sdk_resolver(fx_ver_t version, sdk_roll_forward_policy roll_forwar
...
 
-pal::string_t const& sdk_resolver::global_file_path() const
+const pal::string_t& sdk_resolver::global_file_path() const
 {
     return global_file;
 }
 
+const fx_ver_t& sdk_resolver::requested_version() const
+{
+    return version;
```
Comment (vitek-karas) [inline src/native/corehost/fxr/sdk_resolver.cpp:50]: Nit: could you please rename `version` to `requested_version`? I was looking through the code it if was rather confusing. Also we have some really unfortunate local conflicts, for example: https://github.com/dotnet/runtime/blob/de73d8205047ba946299dc370adbb92f777ab560/src/native/corehost/fxr/sdk_resolver.cpp#L493 this function uses the field at first and then it has a foreach which declares `version` as a local variable.

**Review by vitek-karas — APPROVED** [review]
Nice!

File: src/native/corehost/fxr/sdk_resolver.cpp (hunk 1 above)
Comment (elinor-fung) [reply src/native/corehost/fxr/sdk_resolver.cpp:50]: That is unfortunate indeed 😕

**Review by vitek-karas — APPROVED** [review]
:shipit:

**Comment by elinor-fung** [comment]
Failure is https://github.com/dotnet/runtime/issues/67761

**Pull request merged**